core/domain/platform_feature_services.py

```diff
@@ -149,21 +150,21 @@ def update_feature_flag_rules(
         feature_name, committer_id, commit_message, new_rules)
 
 
-# TODO(#10211): Currently Oppia runs in either of the two modes:
-# dev or prod. There should be another mode 'test' added for QA testing,
-# once it is added, this function needs to be updated to take that into
-# consideration.
 def _get_server_mode() -> platform_parameter_domain.ServerMode:
     """Returns the running mode of Oppia.
 
     Returns:
-        Enum(SERVER_MODES). The server mode of Oppia, dev if Oppia is running
-        in development mode, prod if in production mode.
+        Enum(SERVER_MODES). The server mode of Oppia. This is "dev" if Oppia is
+        running in development mode, "test" if Oppia is running in production
+        mode but not on the main website, and "prod" if Oppia is running in
+        full production mode on the main website.
     """
     return (
         platform_parameter_domain.ServerMode.DEV
         if constants.DEV_MODE
         else platform_parameter_domain.ServerMode.PROD
+        if feconf.OPPIA_PROJECT_ID == 'oppiaserver'
+        else platform_parameter_domain.ServerMode.TEST
```

Comment:
This is the line I mentioned in the mail, @seanlip. The checkpoint features e2e test, while setting up, enables the feature for prod. Setting the env to test, however, causes the flag to be evaluated to false, so the function remains gated.

Reply:
Hi @nithinrdy -- actually, I am a bit confused about that, sorry. If a feature is enabled for prod, wouldn't it be automatically enabled for test as well?

Reply:
Ah, sorry, this was a misinterpretation on my part. It seems that we would need to add multiple rulesets in the admin panel, one for each server mode. (I had thought that prod would automatically "cover" test.)